firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/SetMutation.java

```diff
@@ -72,40 +71,32 @@ public String toString() {
   }
 
   @Override
-  public MaybeDocument applyToRemoteDocument(
-      @Nullable MaybeDocument maybeDoc, MutationResult mutationResult) {
-    verifyKeyMatches(maybeDoc);
+  public void applyToRemoteDocument(Document document, MutationResult mutationResult) {
+    verifyKeyMatches(document);
 
     // Unlike applyToLocalView, if we're applying a mutation to a remote document the server has
     // accepted the mutation so the precondition must have held.
-
-    SnapshotVersion version = mutationResult.getVersion();
-
-    ObjectValue newData = value;
     if (mutationResult.getTransformResults() != null) {
-      List<Value> transformResults =
-          serverTransformResults(maybeDoc, mutationResult.getTransformResults());
-      newData = transformObject(newData, transformResults);
+      Map<FieldPath, Value> transformResults =
```

Comment:
Elsewhere in this PR you changed `getTransformResults` to be non-nullable. That actually seems unrelated to the core of this change, and introduces the risk that it wouldn't be caught in review. Consider decoupling that into a separate change.

Otherwise, the line above needs to be a !`isEmpty()` test.

Reply:
Separated this out as a stand alone PR: https://github.com/firebase/firebase-android-sdk/pull/2428